shin-core/src/format/scenario/info.rs

```diff
@@ -80,34 +115,220 @@ impl MovieInfoItem {
+
+    /// Ends either the list of background definitions at the beginning, or ends an individual character definition, corresponding to a group of outfits (衣装).
+    #[brw(magic = 0x22u8)]
+    EndDefinition,
+}
+pub type CharacterBoxInfo = Vec<CharacterBoxSegment>;
+
+/// An individual instruction for building the data underlying a character in the Characters screen (`chars`).
 #[derive(Debug, BinRead, BinWrite)]
-pub struct Section64InfoItem {
-    pub unk1: U16String,
-    pub unk2: U16List<u16>,
+pub enum CharsSpriteSegment {
+    /// Ends the current character definition.
+    #[brw(magic = 0x0u8)]
+    End,
+
+    #[brw(magic = 0x1u8)]
+    Segment0x1 { unk1: u8 },
+
+    #[brw(magic = 0x2u8)]
+    Segment0x2 {
+        unk1: u8,
+        unk2: u8,
+        unk3: U16String,
+        unk4: U16String,
+    },
+
+    #[brw(magic = 0x3u8)]
+    Segment0x3 { unk1: U16String, unk2: U16String },
+}
+
+impl FinalSegment for CharsSpriteSegment {
+    fn is_final(&self) -> bool {
+        match self {
+            CharsSpriteSegment::End => true,
+            _ => false,
+        }
+    }
 }
-pub type Section64Info = Vec<Section64InfoItem>;
 
+/// The data for a character in the Characters screen (`chars`)
 #[derive(Debug, BinRead, BinWrite)]
-pub struct Section68InfoItem {
-    pub unk1: u16,
-    pub unk2: u16,
-    pub unk3: u16,
+pub struct CharsSpriteInfoItem {
+    /// The episode for which the character sprite and description is valid.
+    pub episode: u8,
+
+    /// The segments defining the sprites and description for this character.
+    #[br(parse_with = parse_terminated_segment_list)]
+    pub segments: Vec<CharsSpriteSegment>,
 }
-pub type Section68Info = Vec<Section68InfoItem>;
 
+pub type CharsSpriteInfo = Vec<CharsSpriteInfoItem>;
+
+/// An individual instruction for building the data underlying the grid in the Characters screen (`chars`).
+#[derive(Debug, BinRead, BinWrite)]
+pub enum CharsGridSegment {
+    /// Ends the current character grid definition.
+    #[brw(magic = 0x0u8)]
+    End,
+
+    /// Defines a portrait on the grid, with a corresponding character to have its full sprite, name, and description shown when the portrait is selected.
+    #[brw(magic = 0x1u8)]
+    Portrait {
+        /// The page to show this portrait on, between 0 and 3 (both inclusive). The game will automatically generate as many pages as necessary for a specific grid.
+        page: u8,
+
+        /// The portrait's X position on the grid.
+        grid_x: u8,
+
+        /// The portrait's Y position on the grid.
+        grid_y: u8,
+
+        /// The ID of the character this portrait is for, indexing into [`CharsSpriteInfo`].
+        character_id: u16,
+
+        /// Defines how the portrait and the corresponding sprite is displayed, and how it behaves on Execute/Resurrect:
+        ///  - **in game**: `0` = empty portrait, `1` = alive, `2` = dead (small blood splotch on sprite), `3` = very dead (large blood splotch on sprite)
```

Comment:
very dead xD

Reply:
Btw, sounds like an ideal case for a enum

Reply:
turns out I was a bit wrong about how exactly this field works, so in the end I didn't end up making an enum here; but I added enums for other relevant fields